server/routerlicious/packages/lambdas/src/scriptorium/lambda.ts

```diff
@@ -69,6 +70,34 @@ export class ScriptoriumLambda implements IPartitionLambda {
 		}
 
 		this.pendingOffset = message;
+
+		if (this.telemetryEnabled) {
+			if (this.pendingMetric === undefined) {
+				// create a new metric for processing the current kafka batch
+				this.pendingMetric = Lumberjack.newLumberMetric(
+					LumberEventName.ScriptoriumProcessBatch,
+					{
+						timestampReadyToProcess: Date.now(),
+						kafkaPartition: this.pendingOffset?.partition,
+						batchOffsetStart: this.pendingOffset?.offset,
+						batchOffsetEnd: this.pendingOffset?.offset,
+						totalBatchSize: boxcar.contents.length,
+					},
+				);
+			} else {
+				// previous batch is still waiting to be processed, update properties in the existing metric
+				this.pendingMetric.setProperty("batchOffsetEnd", this.pendingOffset?.offset);
```

Comment:
Since we are adding and reusing a few new properties that seem to be related/belong to same category, we can also consider adding them here:
https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services-telemetry/src/resources.ts

Reply:
Updated.